src/timeline/widgets/ImageItem.cc

```diff
@@ -61,11 +61,11 @@ ImageItem::ImageItem(QSharedPointer<MatrixClient> client,
 
 ImageItem::ImageItem(QSharedPointer<MatrixClient> client,
                      const QString &url,
-                     const QString &filename,
+                     QSharedPointer<QFile> file,
```

Comment:
How will this handle images that aren't written to disk?

Reply:
I am unsure how to approach this yet as I did not foresee this as a problem. 

To properly load the image, we also need access to the image data. It's not a difficult change, but requires numerous functions to be modified to be able to pass the image data along with the filename. I would like to hear your thoughts on this one.

Reply:
Can you give a few examples of functions that would need to be modified? An image in the process of being uploaded shouldn't need to be accessed from very many places, and I don't imagine there are many places other than the timeline or the upload procedure itself where a filename is pertinent.

Reply:
The following is the calling order which would need to be modified to pass the image data:

1) MatrixClient::imageUploaded
2) TimelineViewManager::queueImageMessage
3) TimelineView::addUserMessage
4) ImageItem constructor

Reply:
That doesn't seem particularly egregious to me. If the increase in number of parameters bothers you, you could always create a `struct Image { QImage data; QString name; };` or similar to bundle them up for convenience.